Makefile

```diff
@@ -1,34 +1,29 @@
-GHCS = 7.8.4 7.10.3 8.0.2 8.2.2
+GHCS = 7.8.4 7.10.3 8.0.2 8.2.2 8.4.1
+
+# default ghc-verison for targets, can be overwritten by make-
+GHC = 8.2.2
+STACK_YAML_ARG = --stack-yaml=stack/stack-$(GHC).yaml
+STACK_CMD = stack $(STACK_YAML_ARG)
 
 .PHONY: all
 all: build test package doc tags
 
-default.nix: extensible-effects.cabal
-	cabal2nix ./. > ./default.nix
-
-.PHONY: init
-init:
-	cabal sandbox init
-	cabal clean
-
 .PHONY: build
-build: init
-	cabal install --only-dependencies --enable-tests --enable-benchmarks
-	cabal configure -flib-Werror --enable-tests --enable-benchmarks -v2 -O2
-	cabal build
+build:
```

Comment:
@power-fungus should `clean` (currently `stack-clean`) be a build pre-requisite for this too?

Reply:
@suhailshergill I would not call it `clean` as it does not clean everything - it only cleans the stack-stuff for the resolver in the default `stack.yaml`. I do not know what would be appropriate for a full `clean`. At least `stack clean` with all GHC-configurations, but maybe deleteing `.stack-work` is the better clean.

Personally, I don't clean before building.

Reply:
note that in `ci-test`, it is cleaned before building

Reply:
@power-fungus i noticed that, hence the suggestion to add it as a pre-requisite for the make target of `build`. in order to be consistent